### PR TITLE
[Serve] JSON logging doc

### LINKED ISSUE #37051: [serve] Document `RAY_SERVE_ENABLE_JSON_LOGGING` feature flag
We should add a section to the logging documentation instructing users on how to enable JSON logging.

This is especially important if we want to make it the default; we should get early feedback first.

### PR DESCRIPTION
**Pull request by sihanwang41**
## Related issue number

Closes: #37051

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/monitoring.md [hunk 1]
```diff
@@ -127,6 +127,17 @@ In addition to the standard Python logger, Serve supports custom logging. Custom
 
 For a detailed overview of logging in Ray, see [Ray Logging](configure-logging).
 
+### JSON logging format
+You can enable JSON logging format in serve log file by setting the environment variable `RAY_SERVE_ENABLE_JSON_LOGGING=1`. After setting this environment variable, the logs will be in the following format:
```
Comment (angelinalg) [inline doc/source/serve/monitoring.md:131]:
```suggestion
You can enable JSON formatted logging in the Serve log file by setting the environment variable `RAY_SERVE_ENABLE_JSON_LOGGING=1`. After setting this environment variable, the logs have the following format:
```

**Review by angelinalg — APPROVED** [review]
Minor suggestion.

File: doc/source/serve/monitoring.md (hunk 1 above)
Comment (edoakes) [reply doc/source/serve/monitoring.md:131]: `JSON-formatted` :)

**Review by edoakes — APPROVED** [review]

File: doc/source/serve/monitoring.md (hunk 1 above)
Comment (shrekris-anyscale) [inline doc/source/serve/monitoring.md:131]: Does this apply to all logs or just the HTTP Proxy logs? The example implies that it's just HTTP Proxy logs.

Comment (sihanwang41) [reply doc/source/serve/monitoring.md:131]: this example is from deployment log not from http proxy. And yes it will enable all serve log files.

**Comment by sihanwang41** [comment]
@edoakes  ready to merge.

**Pull request merged**